src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/S3Configuration.kt

```diff
@@ -1,34 +1,71 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
 
-import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
 import software.amazon.awssdk.regions.Region
 import software.amazon.awssdk.services.s3.S3Client
 import java.net.URI
 
+data class S3Bucket(val enabled: Boolean, val bucketName: String, val client: S3Client)
+
+data class S3BucketConfiguration(
+  val enabled: Boolean,
+  val provider: String,
+  val region: String,
+  val accessKeyId: String,
+  val secretAccessKey: String,
+  val endpoint: Endpoint,
+  val bucket: Bucket,
+) {
+  data class Endpoint(
+    val uri: String
+  )
+
+  data class Bucket(
+    val name: String
+  )
+}
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "aws.s3")
+class S3Buckets(
+  val storage: S3BucketConfiguration,
+  val ndmis: S3BucketConfiguration,
+)
+
 @Configuration
+@EnableConfigurationProperties(S3Buckets::class)
 class S3Configuration(
-  @Value("\${aws.s3.provider}") private val provider: String,
-  @Value("\${aws.s3.region}") private val region: String,
-  @Value("\${aws.s3.access-key-id}") private val accessKeyId: String,
-  @Value("\${aws.s3.secret-access-key}") private val secretAccessKey: String,
-  @Value("\${aws.s3.endpoint.uri:}") private val uri: String,
+  private val buckets: S3Buckets,
 ) {
   @Bean
-  fun s3Client(): S3Client {
+  fun storageS3Bucket(): S3Bucket {
+    return bucketFactory(buckets.storage)
+  }
+
+  @Bean
+  fun ndmisS3Bucket(): S3Bucket {
+    return bucketFactory(buckets.ndmis)
+  }
+
+  private fun bucketFactory(config: S3BucketConfiguration): S3Bucket {
     val builder = S3Client.builder()
-      .region(Region.of(region))
-      .credentialsProvider { AwsBasicCredentials.create(accessKeyId, secretAccessKey) }
+      .region(Region.of(config.region))
+      .credentialsProvider { AwsBasicCredentials.create(config.accessKeyId, config.secretAccessKey) }
```

Comment:
Totally out of bounds of this PR, general 💭 
Can we detect misconfiguration _before_ we attempt to write to the bucket?

Reply:
hm yeh it would be good. i'm not aware of a way to do that, but we could add some kind of health check to the application start up ?